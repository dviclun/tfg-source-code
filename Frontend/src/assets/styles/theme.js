import { createTheme } from "@mui/material";
import { orange } from '@mui/material/colors';
import Poppins from '../fonts/Poppins-Regular.ttf'

export const theme = createTheme({
    status: {
        danger: orange[500],
      },
    palette: {
        slate100: '#f1f5f9',
        slate300:'#cbd5e1',
        green300: '#86efac',
        green400: '#4ade80',
        green500: '#22c55e',
        green600: '#16a34a',
        green700: '#047857', 
        green900: '#14532d',
        red400: '#f87171',
        red500: '#ef4444',
        zinc200: '#e4e4e7',
        zinc300: '#d4d4d8',
        zinc400: '#a1a1aa',
        rose400: '#fb7185',
        rose500: '#f43f5e',
        purple400: '#c084fc',
        purple500: '#a855f7',
        blue300: '#93c5fd',
        blue400: '#60a5fa',
        gray100: '#f3f4f6',
        neutral100: '#f5f5f5',
        neutral200: '#e5e5e5',
        emerald100: '#d1fae5'
    },
});