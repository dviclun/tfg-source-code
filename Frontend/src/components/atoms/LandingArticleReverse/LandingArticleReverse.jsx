import React from 'react'
import { Parallax } from 'react-scroll-parallax';
import { Grid, Typography } from "@mui/material";
import PropTypes from 'prop-types';



export const LandingArticleReverse = ({isSmall, children, image, imageClass, title, ...props}) => {
  return (
    <Grid {...props} >
                    <Grid container item xs={12} md={6} justifyContent='center' alignItems='center'>
                        <Parallax opacity={[0,5]} speed={10}>
                            <Typography variant={isSmall ? 'h6': 'h4'} sx={{textAlign:{xs: 'center', md: 'left'}, fontFamily: 'Poppins', fontWeight: 'bold'}}>{title}</Typography>
                            {children}
                        </Parallax>
                    </Grid>
                    <Grid container item xs={12} md={6} justifyContent='center' alignItems='center'>
                            <Parallax opacity={[0,5]} speed={10}>
                                <img className={imageClass} src={image}></img>
                            </Parallax>    
                    </Grid>  
    </Grid>
  )
}

LandingArticleReverse.propTypes = {
  isSmall: PropTypes.bool,
  children: PropTypes.node,
  image: PropTypes.string,
  imageClass: PropTypes.string,
  title: PropTypes.string,
  props: PropTypes.any
}