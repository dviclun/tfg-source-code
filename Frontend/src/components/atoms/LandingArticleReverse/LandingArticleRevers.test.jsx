import { BrowserRouter } from "react-router-dom";
import { LandingArticleReverse } from "./LandingArticleReverse";
import { AuthProvider } from "../../../contexts/Auth/AuthProvider";
import { ParallaxProvider } from "react-scroll-parallax";
import { ThemeProvider } from "@emotion/react";
import { theme } from "../../../assets/styles/theme";
import i18next from "i18next";
import { I18nextProvider } from "react-i18next";
import { render } from "@testing-library/react";

describe('Test in <LandingArticleReverse/>', () => { 
    test('Container must render and match snapshot', () => { 
        
        const {container} = render(
            <BrowserRouter>
                <AuthProvider>
                    <ParallaxProvider>
                        <ThemeProvider theme={theme}>
                        <I18nextProvider i18n={i18next}>
                            <LandingArticleReverse/>    
                        </I18nextProvider>
                        </ThemeProvider>
                    </ParallaxProvider>
                </AuthProvider>
            </BrowserRouter>
        )

        expect(container).toBeTruthy();
        expect(container).toMatchSnapshot();
     })
 })