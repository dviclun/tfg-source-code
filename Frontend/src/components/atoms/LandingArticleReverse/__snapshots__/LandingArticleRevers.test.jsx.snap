// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <LandingArticleReverse/> Container must render and match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root css-vj1n65-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-lpp2qy-MuiGrid-root"
    >
      <div
        style="will-change: transform,opacity; transform: translateY(-100px); opacity: 5;"
      >
        <h4
          class="MuiTypography-root MuiTypography-h4 css-14atv5i-MuiTypography-root"
        />
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-md-6 css-lpp2qy-MuiGrid-root"
    >
      <div
        style="will-change: transform,opacity; transform: translateY(-100px); opacity: 5;"
      >
        <img />
      </div>
    </div>
  </div>
</div>
`;
